File: nom-api/Nom.Orch/Models/Measurement/MeasurementCategoryModel.cs
namespace Nom.Orch.Models.Measurement
{
    /// <summary>
    /// Model representing a measurement category in the API layer.
    /// </summary>
    public class MeasurementCategoryModel
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public long BaseUnitId { get; set; }
        public string BaseUnitName { get; set; } = string.Empty;
        public string BaseUnitSymbol { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; }
        public DateTime? LastModifiedDate { get; set; }
    }
}

File: nom-ui/src/app/measurement/models/measurement-category.model.ts
export interface MeasurementCategoryModel {
    id: number;
    name: string;
    description?: string;
    baseUnitId?: number;
    baseUnitName?: string;
    baseUnitSymbol?: string;
    createdDate: Date;
    lastModifiedDate?: Date;
    authorId: number;
    createdById: number;
    userId: number;
}
